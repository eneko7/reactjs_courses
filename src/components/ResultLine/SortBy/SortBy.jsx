import React from 'react';
import SortInput from '../../SortInput/SortInput';
import styles from './SortBy.scss';

const SortBy = () => (
  <div className={styles.sortBy}>
    <span className={styles.sortBy_span}>Sort by</span>
    <SortInput
      name="sortBy"
      id="releaseDate"
      text="Release date"
      inputClass="sortBy_input"
      labelClass="sortBy_button"
      defaultChecked="defaultChecked"
    />
    <SortInput
      name="sortBy"
      id="Rating"
      text="Rating"
      inputClass="sortBy_input"
      labelClass="sortBy_button"
    />
  </div>
);

export default SortBy;
