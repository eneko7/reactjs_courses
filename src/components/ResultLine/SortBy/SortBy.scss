@import "../../../assets/constants/constants.scss";

.sort_by {
  display: flex;
  justify-content: space-between;
  width: 25%;
}

.sort_by__span {
  font-family: $fontFamilyBold;
  font-size: 20px;
}

@media (max-width: $desktop) {
  .sort_by {
    width: 30%;
  }
}

@media (max-width: $laptop) {
  .sort_by {
    width: 40%;
  }
}

@media (max-width: $tablet) {
  .sort_by {
    width: 48%;
  }
}

@media (max-width: $phone) {
  .sort_by {
    width: 90%;
    margin-top: 15px;
  }
}
