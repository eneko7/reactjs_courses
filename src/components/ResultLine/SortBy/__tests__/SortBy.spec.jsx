import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SortBy from '../SortBy';

const renderer = new ShallowRenderer();
renderer.render(<SortBy />);
describe('SortBy', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
