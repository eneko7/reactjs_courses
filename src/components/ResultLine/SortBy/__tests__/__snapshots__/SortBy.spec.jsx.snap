// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy renders correctly 1`] = `
<div
  className="sortBy"
>
  <span
    className="sortBy_span"
  >
    Sort by
  </span>
  <SortInput
    defaultChecked="defaultChecked"
    id="releaseDate"
    inputClass="sortBy_input"
    labelClass="sortBy_button"
    name="sortBy"
    text="Release date"
  />
  <SortInput
    defaultChecked=""
    id="Rating"
    inputClass="sortBy_input"
    labelClass="sortBy_button"
    name="sortBy"
    text="Rating"
  />
</div>
`;
